### ISSUE TITLE
"giant rendering overhaul and many small fixes" Not building

### ISSUE THREAD

**Issue by alcomposer** [olive-editor/olive#553]
Latest git commit: giant rendering overhaul and many small fixes
OS: Arch Linux
```
In file included from ../olive/project/sourcescommon.cpp:36:
../olive/ui/menuhelper.h:123:17: error: expected ‘;’ at end of member declaration
   static QMenu* create_submenu(QMenuBar* parent,
                 ^~~~~~~~~~~~~~
                               ;
../olive/ui/menuhelper.h:123:40: error: expected ‘)’ before ‘*’ token
   static QMenu* create_submenu(QMenuBar* parent,
                               ~        ^
                                        )
../olive/ui/menuhelper.h:126:45: error: ‘static QMenu* MenuHelper::create_submenu(QMenu*)’ conflicts with a previous declaration
   static QMenu* create_submenu(QMenu* parent);
                                             ^
../olive/ui/menuhelper.h:123:17: note: previous declaration ‘QMenu* MenuHelper::create_submenu’
   static QMenu* create_submenu(QMenuBar* parent,
                 ^~~~~~~~~~~~~~
make: *** [Makefile:4496: sourcescommon.o] Error 1
```

**Comment by alcomposer** [comment]
I was able to get this building again: https://github.com/alcomposer/olive/commits/not_rendering
However when I build that branch, Olive runs but doesn't show Sequence Viewer (its black)

@itsmattkc sorry for messing up your name in commit message- I wasn't intending to share this! :-(

**Comment by itsmattkc** [comment]
Ahh thank you, I'd noticed it wasn't compiling anymore in GCC but couldn't figure out why. I've implemented your fix in da2f51ee6e6d808891a1370bfd2b6c54769e8caa and it seems to be working now.

**Comment by alcomposer** [comment]
I still get this in GCC:
```
effects/internal/frei0reffect.cpp:209:36: error: ‘using element_type = class Clip’ {aka ‘class Clip’} has no member named ‘getWidth’
  instance = construct(parent_clip->getWidth(), parent_clip->getHeight());
                                    ^~~~~~~~
effects/internal/frei0reffect.cpp:209:61: error: ‘using element_type = class Clip’ {aka ‘class Clip’} has no member named ‘getHeight’; did you mean ‘media_height’?
  instance = construct(parent_clip->getWidth(), parent_clip->getHeight());
```

**Comment by itsmattkc** [comment]
Ahh yes sorry, I missed your other change. I've pushed that now too in 39d62fe39486526ecaca7ace3006407107c270df

**Comment by alcomposer** [comment]
In my previous comment above: https://github.com/olive-editor/olive/issues/553#issuecomment-467317188

I am able to then build, but there are still problems. GCC builds, but segfaults.

**Comment by itsmattkc** [comment]
I see, I'll have to check that out. Thanks for letting me know

**Comment by alcomposer** [comment]
GCC build segfaults, but QtCreator runs - but doesn't show sequence viewer

**Comment by alcomposer** [comment]
Hmm, ok:

I built Olive in GCC with debug & gdb symbols = no segfault at startup

**Comment by itsmattkc** [comment]
Something exclusive to release builds? It may be something behind the `QT_DEBUG` #define making a difference? Olive uses those in a few places.

**Comment by alcomposer** [comment]
Ok, I will see if adding -o by itself will cause a segfault.

**Comment by alcomposer** [comment]
Ok, I get this in gdb:

```
(gdb) start
Temporary breakpoint 1 at 0x512d0
Starting program: /home/alexmitchell/Documents/github/olive/olive-editor 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".

Temporary breakpoint 1, 0x00005555555a52d0 in main ()
(gdb) next
Single stepping until exit from function main,
which has no line number information.
[New Thread 0x7fffea875700 (LWP 31217)]
2019-02-26T17:46:56 [DEBUG] using qt5ct plugin
2019-02-26T17:46:56 [WARNING] QApplication: invalid style override passed, ignoring it.
    Available styles: bb10dark, bb10bright, cleanlooks, gtk2, cde, motif, plastique, qt5ct-style, Windows, Fusion
2019-02-26T17:46:56 [WARNING] libpng warning: iCCP: known incorrect sRGB profile
2019-02-26T17:46:56 [INFO] Initializing custom cursors
2019-02-26T17:46:56 [INFO] Finished initializing custom cursors
[New Thread 0x7fffe8853700 (LWP 31218)]
[New Thread 0x7fffe2395700 (LWP 31219)]
[New Thread 0x7fffe1b94700 (LWP 31220)]
[New Thread 0x7fffe12e6700 (LWP 31221)]
2019-02-26T17:46:57 [INFO] Initializing effects...
2019-02-26T17:46:57 [DEBUG] D-Bus global menu: no
2019-02-26T17:46:58 [INFO] Finished initializing effects
[Thread 0x7fffe12e6700 (LWP 31221) exited]
[New Thread 0x7fffe12e6700 (LWP 31222)]
[New Thread 0x7fffbffff700 (LWP 31223)]
[New Thread 0x7fffb77fe700 (LWP 31224)]

Thread 1 "olive-editor" received signal SIGSEGV, Segmentation fault.
0x00007ffff5297600 in operator==(QString const&, QString const&) () from /usr/lib/libQt5Core.so.5
```

**Comment by itsmattkc** [comment]
The segfault should be fixed as of 3233c0d0f96210b973e485911c4bc84103c593ed.
